feat: Improving conversion goal experience

jabahamondes opened the pull request:
## Problem

We have a very ugly and not the best experience for trying a new conversion goal.

## Changes

Since it's not only related to the table component but also the aggregated results, I'm moving this into a filter on top.

## How did you test this code?
Manually

Filter:
<img width="1276" height="76" alt="image" src="https://github.com/user-attachments/assets/c3d4b9e4-1d03-4769-98e3-ccb29113b83d" />


Modal:
<img width="556" height="362" alt="Screenshot 2025-11-28 at 7 01 05 PM" src="https://github.com/user-attachments/assets/951621b5-c62e-4089-a7e2-6b4f46148c6f" />

Now without:
<img width="1255" height="198" alt="image" src="https://github.com/user-attachments/assets/ed774441-2043-4d61-a0ac-93c7e23d2afd" />


Before:
<img width="1261" height="343" alt="image" src="https://github.com/user-attachments/assets/979adf0e-85dd-45f0-864b-faf3d839d13e" />

jabahamondes committed:
Improving conversion goal experience

jabahamondes committed:
Removing some duplcated text

jabahamondes committed:
Fixing some bugs

jabahamondes committed:
Fixing css

adboio commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/MarketingAnalyticsFilters/ConversionGoalFilterButton.tsx:
```diff
@@ -0,0 +1,58 @@
+
+    const handleSaveConversionGoal = useCallback((): void => {
+        addOrUpdateConversionGoal({ ...conversionGoalInput, conversion_goal_name: uniqueConversionGoalName })
+        saveDraftConversionGoal()
```
it could just be that i'm still learning kea (sadge), but this feels a little confusing - saveDraftConversionGoal is basically a no-op with a listener that calls resetConversionGoalInput, which has reducers that reset the state, but also a listener that calls setDraftConversionGoal(null)  
  
not blocking, but feels a bit redundant and could maybe be simplified (?)

jabahamondes replied:
Yes, I did this before because I wanted to separate the concepts: resetConversionGoalInput vs when saveDraftConversionGoal is called. I will still refactor all of this a little bit to make it more clear.

adboio commented on frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/MarketingAnalyticsFilters/ConversionGoalModal.tsx:
[sorta nit] the `handleXXX`​ functions here feel a bit verbose/redundant, i imagine the logic could be updated to handle a lot of this orchestration instead of manually sequencing multiple actions, like

```
addOrUpdateConversionGoal({ ...conversionGoalInput, conversion_goal_name: uniqueConversionGoalName })
saveDraftConversionGoal()
hideConversionGoalModal()
```

could maybe just be like a `saveConversionGoal`​ action with a listener - this would also prevent orchestration logic duplication between the modal and the button

adboio reviewed (approved):
couple nits but no big deal - lgtm!

jabahamondes committed:
PRs comments: Refactor conversion goal handling in Marketing Analytics

- Simplified the logic for saving and clearing conversion goals in ConversionGoalFilterButton and ConversionGoalModal components.
- Removed unnecessary callbacks and integrated direct action calls for better readability and maintainability.
- Updated the UI to reflect these changes, ensuring a smoother user experience when managing conversion goals.

This improves the overall conversion goal experience and aligns with recent enhancements in the marketing analytics settings.

jabahamondes committed:
Merge branch 'master' into feat/moving-out-the-exploring-conversion-goal-input
